### PR TITLE
fix(AS): fix AS group resource issues of codes, documents and test cases.

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:

fix AS group resource issues of codes, documents and test cases.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

- commit1: Fix AS group resource lint error.
- commit2: Fix AS group document error.
- commit3: Fix AS group test cases error

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccASGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccASGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccASGroup_basic
=== PAUSE TestAccASGroup_basic
=== CONT  TestAccASGroup_basic
--- PASS: TestAccASGroup_basic (341.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        341.905s

make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccASGroup_withEpsId'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccASGroup_withEpsId -timeout 360m -parallel 4
=== RUN   TestAccASGroup_withEpsId
=== PAUSE TestAccASGroup_withEpsId
=== CONT  TestAccASGroup_withEpsId
--- PASS: TestAccASGroup_withEpsId (170.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        170.844s

make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccASGroup_forceDelete'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccASGroup_forceDelete -timeout 360m -parallel 4
=== RUN   TestAccASGroup_forceDelete
=== PAUSE TestAccASGroup_forceDelete
=== CONT  TestAccASGroup_forceDelete
--- PASS: TestAccASGroup_forceDelete (235.76s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        235.804s

make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccASGroup_sourceDestCheck'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccASGroup_sourceDestCheck -timeout 360m -parallel 4
=== RUN   TestAccASGroup_sourceDestCheck
=== PAUSE TestAccASGroup_sourceDestCheck
=== CONT  TestAccASGroup_sourceDestCheck
--- PASS: TestAccASGroup_sourceDestCheck (158.62s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        158.667s
```

### CODE REVIEW

File: docs/resources/as_group.md [hunk 1]
```diff
@@ -145,18 +123,19 @@ The following arguments are supported:
 * `desire_instance_number` - (Optional, Int) Specifies the expected number of instances. The default value is the
   minimum number of instances. The value ranges from the minimum number of instances to the maximum number of instances.
 
-* `min_instance_number` - (Optional, Int) Specifies the minimum number of instances. The default value is 0.
+* `min_instance_number` - (Optional, Int) Specifies the minimum number of instances. Defaults to **0**.
```
Comment (jinyangyang222) [inline docs/resources/as_group.md:126]: Defaults to `0`.
number use (``).

File: docs/resources/as_group.md [hunk 2]
```diff
@@ -145,18 +123,19 @@ The following arguments are supported:
...
 
-* `max_instance_number` - (Optional, Int) Specifies the maximum number of instances. The default value is 0.
+* `max_instance_number` - (Optional, Int) Specifies the maximum number of instances. The value ranges from **0** to **300**.
+  Defaults to **0**.
 
-* `cool_down_time` - (Optional, Int) Specifies the cooling duration (in seconds). The value ranges from 0 to 86400,
-  and is 300 by default.
+* `cool_down_time` - (Optional, Int) Specifies the cooling duration (in seconds). The value ranges from **0** to **86400**.
```
Comment (jinyangyang222) [inline docs/resources/as_group.md:131]: The value ranges from `0` to `86,400`.

File: docs/resources/as_group.md (hunk 1 above)
Comment (jinyangyang222) [reply docs/resources/as_group.md:126]: Please also check the relevant formats in other places

File: docs/resources/as_group.md (hunk 2 above)
Comment (jinyangyang222) [reply docs/resources/as_group.md:131]: Please also check the relevant formats in other places

File: docs/resources/as_group.md [hunk 3]
```diff
@@ -204,36 +184,41 @@ The following arguments are supported:
...
 * `agency_name` - (Optional, String) Specifies the IAM agency name. If you change the agency,
   the new agency will be available for ECSs scaled out after the change.
 
-* `delete_publicip` - (Optional, Bool) Specifies whether to delete the elastic IP address bound to the instances of
-  AS group when deleting the instances. The options are `true` and `false`.
+* `delete_publicip` - (Optional, Bool) Specifies whether to release the EIPs bound to ECSs when the ECSs are removed
+  from the AS group. Defaults to **false**.
+
+  + **true**: The EIPs bound to ECSs will be released when the ECSs are removed. If the EIPs are billed on a
+    `yearly/monthly` basis, they will not be released when the ECSs are removed.
```
Comment (jinyangyang222) [inline docs/resources/as_group.md:196]: **yearly/monthly**, use (** **)

File: docs/resources/as_group.md (hunk 1 above)
Comment (deer-hang) [reply docs/resources/as_group.md:126]: keep status quo

File: docs/resources/as_group.md (hunk 2 above)
Comment (deer-hang) [reply docs/resources/as_group.md]: OK

File: docs/resources/as_group.md (hunk 3 above)
Comment (deer-hang) [reply docs/resources/as_group.md:196]: keep status quo

**Pull request merged**